challenge_8/python/mjuiuc/solution.py

```diff
@@ -0,0 +1,23 @@
+from listNode import listNode
+
+
+def solution(head):
+	lop = dict()
+	newHead = listNode()
+	temp = head
+	temp2 = newHead
+	# copy the list first
+	while temp != None:
+		temp2.val = temp.val
+		lop[temp2.val] = temp2
```

Comment:
This makes it such that you cannot have duplicate numbers in your list.  `LinkedLists` should allow duplicate values, otherwise you've made a `LinkedSet`